omegaconf/dictconfig.py

```diff
@@ -377,6 +375,7 @@ def __getitem__(self, key: DictKeyType) -> Any:
             self._format_and_raise(key=key, value=None, cause=e)
 
     def __delitem__(self, key: DictKeyType) -> None:
+        key = self._validate_and_normalize_key(key)
```

Comment:
This looks like a bugfix. It would be better to include it in a standalone PR but I will not insist here.

1. Do you have a test covering this?
2. Add a news fragment?

Reply:
>     1. Do you have a test covering this?
There is some test coverage in the new `test_compare_dictconfig_vs_dict.py` file.

Reply:
Specifically the test coverage is in the `test__delitem__enum_typed` and `test__delitem__enum_typed_struct_mode` methods.

Reply:
I created an issue to track the bug: #554

Reply:
And here is the news fragment: https://github.com/omry/omegaconf/pull/484/commits/91e13a94d1166da2501becb14c79df2b93b2b287